pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -969,15 +965,19 @@ func (cm *containerManagerImpl) GetDynamicResources(pod *v1.Pod, container *v1.C
 		// a set of CDIDevices from a different kubelet plugin. In the future we may want to
 		// include the name of the kubelet plugin and/or other types of resources that are
 		// not CDIDevices (assuming the DRAmanager supports this).
-		for _, klPluginCdiDevices := range containerClaimInfo.CDIDevices {
+		for /* driverName */ _, driverState := range containerClaimInfo.DriverState {
 			var cdiDevices []*podresourcesapi.CDIDevice
-			for _, cdiDevice := range klPluginCdiDevices {
-				cdiDevices = append(cdiDevices, &podresourcesapi.CDIDevice{Name: cdiDevice})
+			for _, device := range driverState.Devices {
+				for _, cdiDeviceID := range device.CDIDeviceIDs {
+					cdiDevices = append(cdiDevices, &podresourcesapi.CDIDevice{Name: cdiDeviceID})
+				}
+			}
+			resources := &podresourcesapi.ClaimResource{
+				CDIDevices: cdiDevices,
 			}
-			claimResources = append(claimResources, &podresourcesapi.ClaimResource{CDIDevices: cdiDevices})
+			claimResources = append(claimResources, resources)
 		}
```

Comment:
This needs to be reworked to build one ClaimResource object (and populate it) with each unique combination of (driverName, poolName, deviceName, cdiDevices).

Reply:
I see that initializing `driverName, poolName, deviceName` is missing above in `podresourcesapi.ClaimResource{}`. Once I add it, is that sufficient?

Reply:
Not quite. There should be a separate claimResource for each allocated device.

So something like:
```
		for /* driverName */ _, driverState := range containerClaimInfo.DriverState {
			for _, device := range driverState.Devices {
				var cdiDevices []*podresourcesapi.CDIDevice
				for _, cdiDeviceID := range device.CDIDeviceIDs {
					cdiDevices = append(cdiDevices, &podresourcesapi.CDIDevice{Name: cdiDeviceID})
				}
				resource := &podresourcesapi.ClaimResource{
					CDIDevices: cdiDevices,
					DriverName: ...,
					PoolName: ...,
					DeviceName: ...,
				}
				claimResources = append(claimResources, resource)
			}
		}
```

One question I have though, is if `DeviceName` should be renamed to `ResourceName` since now we have devices, but in the future we may have gizmos....

Reply:
> Not quite. There should be a separate claimResource for each allocated device.

Ah, right.

> One question I have though, is if DeviceName should be renamed to ResourceName since now we have devices, but in the future we may have gizmos....

We don't have a "ResourceName". If we add gizmos, let's add a `GizmoName` alongside `DeviceName` instead of using the same field for two different things. That way, a consumer can also see what the entry is about.